### PR TITLE
Fix bugs in the Case translation in Core-to-Geb

### PR DESCRIPTION
**Pull request by lukaszcz**

### CODE REVIEW

File: src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs [hunk 1]
```diff
@@ -120,7 +120,7 @@ fromCore tab = case tab ^. Core.infoMain of
         nodeType = convertType (Info.getNodeType node)
 
     convertNode :: Core.Node -> Trans Morphism
-    convertNode = \case
+    convertNode node = case node of
```
Comment (jonaprieto) [inline src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs:123]: If you're not using node, please leave it as it was.

File: src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs [hunk 2]
```diff
@@ -414,17 +416,33 @@ fromCore tab = case tab ^. Core.infoMain of
...
           if
               | _caseBranchBindersNum == 0 -> return branch
-              | otherwise -> do
-                  val' <- convertNode val
-                  return $ mkApps (mkLambs branch argtys) val' valty argtys
+              | _caseBranchBindersNum == 1 ->
+                  return $
+                    MorphismApplication
+                      Application
+                        { _applicationDomainType = valty,
```
Comment (jonaprieto) [inline src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs:428]: can you keep the pattern, here it'd be: "valType" or "domainType".

**Comment by jonaprieto** [comment]
The files in tests/Geb/positive/out/ should have the .geb extension.

File: src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs (hunk 1 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs:123]: Done

File: src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs (hunk 2 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs:428]: Done

**Comment by lukaszcz** [comment]
> The files in tests/Geb/positive/out/ should have the .geb extension.

Both in Eval/out and Infer/out we use the .out extension. If we want to change it, let's change all of them in the future PR which will introduce automated tests for the Core-to-GEB translation.

**Review by jonaprieto — APPROVED** [review]

**Pull request merged**